src/super_gradients/training/transforms/transforms.py

```diff
@@ -743,6 +755,33 @@ def get_equivalent_preprocessing(self):
         raise NotImplementedError("get_equivalent_preprocessing is not implemented for non-deterministic transforms.")
 
 
+@register_transform(Transforms.SegToTensor)
+class SegToTensor(AbstractSegmentationTransform, LegacySegmentationTransformMixin):
+    def apply_to_sample(self, sample: SegmentationSample) -> SegmentationSample:
+        sample.image = to_tensor(sample.image)
+        sample.mask = torch.from_numpy(np.array(sample.mask)).long()
```

Comment:
Let's check that this transform can be used with binary segmentation (BCE loss) and semantic segmentation (CE loss).
A few pitfalls that I'm aware of:
- For BCE the shape must match the shape of preds (BCHW) and also targets should be of float format (and not long, nn.BCEWithLogistLoss will throw an error if target is long if I'm not mistaken)
- OpenCV may silently squeeze channel dimension (HW1 -> HW) in resize/pad/etc. In Albu it is handled by `@preserve_channel_dim` decorator. We have want to have similar decorator OR have the argument in `SegToTensor(dummy_channel_dim=True)` to explicitly tell whether dummy channel dim is needed.
- Maybe same of output dtype in `SegToTensor(mask_dtype=...)`

Reply:
added 713401666e053c8196a8e979b6d6c485167d25d3